backend/corpora/common/dataset_validator.py

```diff
@@ -40,21 +43,88 @@ def validate_dataset_file(self):
 
     def validate_h5ad_dataset(self, file_object):
         """
-        Reads the file contents into an AnnData object. Each attribute of the AnnData object will then be checked to
-        ensure it contains the appropriate metadata.
+        Reads the H5AD file contents into an AnnData object. Each attribute of the AnnData object will then be
+        checked to ensure it contains the appropriate metadata.
         """
 
         start_time = time.time()
-        logging.info("Reading anndata object...")
+        logging.info("Reading H5AD file into anndata object...")
         anndata_object = anndata.read_h5ad(file_object)
         logging.info(f"Finished reading anndata object in {time.time() - start_time:.3f} seconds.")
 
+        self.validate_anndata_object(anndata_object)
+
+    def validate_loom_dataset(self, file_object, loom_x_layer_name=None):
+        """
+        Reads the Loom file contents into an AnnData object. Each attribute of the AnnData object will then be
+        checked to ensure it contains the appropriate metadata.
+        """
+
+        start_time = time.time()
+        logging.info("Reading Loom file into anndata object...")
+        if loom_x_layer_name:
+            anndata_object = anndata.read_loom(file_object, X_name=loom_x_layer_name)
+        else:
+            anndata_object = anndata.read_loom(file_object)
+        logging.info(f"Finished reading anndata object in {time.time() - start_time:.3f} seconds.")
+
+        self.validate_anndata_object(anndata_object)
+
+    def validate_anndata_object(self, anndata_object: anndata.AnnData):
         start_time = time.time()
-        logging.info("Beginning validation of file...")
+        logging.info("Beginning validation of anndata object...")
+        self.verify_layers(anndata_object)
         self.verify_obs(anndata_object)
         self.verify_vars(anndata_object)
         self.verify_uns(anndata_object)
-        logging.info(f"Finished completing validation on the file in {time.time() - start_time:.3f} seconds.")
+        logging.info(f"Finished completing validation in {time.time() - start_time:.3f} seconds.")
+
+    def verify_layers(self, data_object: anndata.AnnData):
+        """
+        Verifies that the dataset contains at least the raw data and if other layers are provided, that they each
+        contain an appropriate description.
+        """
+
+        # Check to make sure X data exists
+        has_data = True
+        if isinstance(data_object.X, DataFrame):
+            has_data = data_object.X.data.any()
+        elif isinstance(data_object.X, ndarray):
+            has_data = data_object.X.any()
+        elif isinstance(data_object.X, spmatrix):
+            has_data = (data_object.X.count_nonzero() == data_object.X.nnz) or data_object.X.nnz == 0
+        else:
+            logging.warning(
+                f"Could not check raw data layer to ensure that it exists. The type is " f"{type(data_object.X)}!"
```

Comment:
I believe `raw` means something special here, so maybe use a different word?

Reply:
Lol, yeah you're right. Don't know what I was thinking - changed to X.